### ISSUE TITLE
Soft-Clicker Server

### ISSUE THREAD

**Issue by chamika** [charithag/soft-clicker#2]
We need to finalize the features provided with the Soft-Clicker Server. As far as I understood, this would be the application which is used by the lecturer. So the lecturer should have following features.
- Enable service discovery mode so that students can identify the IP of the lecturer's device
- Enable Quiz mode so that students can submit the answers
- Capture students' answers and store them in a database 
- Ability to manage quizzes
- View answers of the students

**Capture students' answers and store them in a database**
Quizzes should be organized in this manner.

```
Course (CS5440 - WAN)
     |-- Class (2016)
     |-- Class (2015)
            |-- Quiz 1
                  |-- Question 1
                  |-- Question 2
            |-- Quiz 2
```

Under the Quiz we should store the answers of the students.

**Ability to manage quizzes**
Lecturer should have following features
- Create Courses
- Create Classes
- Create Quizzes
- Setup Questions and answers for the quizzes

**View answers of the students**
- View for answer count
- Graphical representation of answers (Graphs like pie-charts)
- Methods to compare quizzes with other Classes of the same Course

**Comment by chamika** [comment]
For DB implementation, we can have a simple one like SQLite DB which creates a relational database in a single file. Since the server is also a Java application, in order to run it directly starting a DB server is not necessary.

For the Java application UI [JFoenix](https://github.com/jfoenixadmin/JFoenix) will give more material design look which will tally with the Android application.

**Comment by charithag** [comment]
+1 For the idea discussed. Anyway I think it is better to h2 DB for server rather than SQLite, since has few advantages over SQLite[1].

[1] http://www.webdevelopersdiary.com/blog/lightweight-self-contained-database-sqlite-vs-h2-database-vs-mysql-embedded

**Comment by chamika** [comment]
Since this is a simple application which has small data volume I don't think there will be a considerable advantage over SQLite. @charithag Have you tried integrating h2DB with JDBC?

**Comment by charithag** [comment]
Yes. Got your point. BTW, it is very easy to connect with embedded h2 with only few lines of code as same as we are doing it in MySQL[1]. Anyway if it is more convenient to use SQLite, I'm also +1 for it.

[1] http://www.h2database.com/html/faq.html#connect

**Comment by rasika** [comment]
I analysed the comparison provided by charithag. Seems like H2 provides OOB support for many features. Since we don't have a requirement to have the minimum footprint and compatibility to other DBMS(later lecturer may be able to plug in another external DBMS without changing a single bit in the code) I am +1 for H2.

**Comment by RashmikaW** [comment]
I'm going to create UI's for managing quizzes on the server.

**Comment by rasika** [comment]
@RashmikaW please create a separate issue to discuss on UI bits.
